### PR TITLE
[V2] chore:  merge V2 upstream as of 2023-02-13

### PR DESCRIPTION
**Pull request by sunpa93**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by sunpa93** [comment]
TODO: assess whether v2 driver can / should support `allow-empty-cloud-config` mode

File: pkg/azuredisk/azuredisk_v2.go [hunk 1]
```diff
@@ -217,51 +218,53 @@ func (d *DriverV2) Run(endpoint, kubeconfig string, disableAVSetNodes, testingMo
...
-			d.config.CloudConfig,
-			d.isPerfOptimizationEnabled(),
-			topologyKey,
-			userAgent,
-			d.config.ControllerConfig.EnableDiskOnlineResize,
-			d.config.ControllerConfig.EnableAsyncAttach)
-		if err != nil {
-			klog.Fatalf("Failed to get controller provisioner. Error: %v", err)
+		// d.cloudProvisioner is set by NewFakeDriver for unit tests.
+		if d.cloudProvisioner == nil {
```
Comment (edreed) [inline pkg/azuredisk/azuredisk_v2.go:223]: Why make initialization of `cloudProvisioner` dependent on `crdProvisioner == nil`?

Comment (sunpa93) [reply pkg/azuredisk/azuredisk_v2.go:223]: It seems I have missed a curly brace as I was reverting some changes. Good catch :)

File: test/e2e/manifest/migration-windows.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
         "windowsProfile": {
             "adminUsername": "azureuser",
             "adminPassword": "replacepassword1234$",
-            "csiProxyURL": "https://acs-mirror.azureedge.net/csi-proxy/v1.0.2/binaries/csi-proxy-v1.0.2.tar.gz",
+            "csiProxyURL": "https://acs-mirror.azureedge.net/csi-proxy/v0.2.2/binaries/csi-proxy-v0.2.2.tar.gz",
```
Comment (edreed) [inline test/e2e/manifest/migration-windows.json:44]: Odd, I wonder why this was changed to the pre GA version of CSI Proxy?

Comment (sunpa93) [reply test/e2e/manifest/migration-windows.json:44]: I see that this change was made 3 months ago https://github.com/kubernetes-sigs/azuredisk-csi-driver/commit/e3d0c08f4d95baf4c9a6c5906f45e705567f6e13. Maybe I am missing something but I can't find a PR for this.

**Pull request merged**